### ISSUE TITLE
3.4.0-beta1: Argument 2 passed to Cake\Controller\Controller::__construct() error

### ISSUE THREAD

**Issue by samsonasik** [cakephp/cakephp#9949]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.4.0-beta1
* Platform and Target: Apache/2.4.16, PHP 7.1

### What I did
- composer create-project --prefer-dist cakephp/app cakeapp
- composer require "cakephp/cakephp:3.4.0-beta1"
- bin/cake server

### What happened
Got Error:
```
 Argument 2 passed to Cake\Controller\Controller::__construct() must be an instance of Cake\Network\Response or null, instance of Cake\Http\Response given
```
![screen shot 2016-12-31 at 5 54 02 pm](https://cloud.githubusercontent.com/assets/459648/21577122/277fc61c-cf82-11e6-8505-6829448a63bc.png)

### What you expected to happen
Should show home page.

**Comment by luke83** [comment]
i had the same problem, turning off debug (and so debug_kit) solved in my case, i think it is a debug_kit issue (cakephp/debug_kit#469)

@samsonasik may you try to turn off debug in config/app.php? this should not load debug_kit plugin and avoid the warnings..

**Comment by dereuromark** [comment]
Also, make sure you cleared the cache.

**Comment by samsonasik** [comment]
cache cleared, turn off `debug` config at `config/app.php`:
```php
// config/app.php
 'debug' => false,
```
got:
```
An Internal Error Has Occurred
```
error

**Comment by ADmad** [comment]
@samsonasik Keep debug on, just comment out debugkit loading in your bootstrap.

**Comment by ADmad** [comment]
Also please use the `3.next` branch of `cakephp/app` to test.

**Comment by samsonasik** [comment]
I am now using 3.next branch:
```
rm -rf cakeapp
composer create-project --prefer-dist cakephp/app:dev-3.next cakeapp
composer require "cakephp/cakephp:3.4.0-beta1"
bin/cake server
```
 got same error, even with comment debugkit loading:
```php
if (Configure::read('debug')) {
//    Plugin::load('DebugKit', ['bootstrap' => true]);
}
```
Got same error

**Comment by luke83** [comment]
@ADmad the warnings are also in the 3.next: https://travis-ci.org/cakephp/app/jobs/187886574 (please look the job log)

**Comment by ADmad** [comment]
#9946 fixes this issue. Please wait until it's merged and then try `3.next` of cakephp core.

**Comment by dereuromark** [comment]
Is that a E_STRICT error? That could be silenced then for the time being afaik.

**Comment by flashios09** [comment]
it says:
```
Error in: ROOT/vendor/cakephp/cakephp/src/Controller/Controller.php, line 232
```
if i change the use response from `Cake\Network\Response` to `Cake\Http\Response` it works

**Comment by dereuromark** [comment]
Does
    
    error_reporting(E_ALL & ~E_STRICT);

temporary silence it?

**Comment by flashios09** [comment]
should we use `Cake\Network\Response` or `Cake\Http\Response` ?

**Comment by ADmad** [comment]
@flashios09  You don't need to make any change. Just wait for the fix I linked earlier to be merged.

**Comment by markstory** [comment]
This is fixed by #9946
